.circleci/config.yml

```diff
@@ -0,0 +1,129 @@
+version: 2.1
+
+workflows:
+  build_and_test:
+    jobs:
+      - build
+      - jenkins_test:
+          jenkins_version: lts
+
+  nightly_jenkins_test:
+    triggers:
+      - schedule:
+          cron: "0 6 * * *"
+          filters:
+            branches:
+              only:
+                - master
+    jobs:
+      - jenkins_test:
+          jenkins_version: lts
+      - jenkins_test:
+          jenkins_version: latest
+
+jobs:
+  build:
+    docker:
+      - image: circleci/openjdk:8u222-jdk-stretch
+
+    working_directory: ~/repo
+
+    environment:
+      # Customize the JVM maximum heap limit
+      MAVEN_OPTS: -Xmx3200m
+
+    steps:
+      - checkout
+
+      # Download and cache dependencies
+      - restore_cache:
+          keys:
+            - v1-dependencies-{{ checksum "pom.xml" }}
+            # fallback to using the latest cache if no exact match is found
+            - v1-dependencies-
+
+      - run: mvn -B dependency:go-offline
+
+      - save_cache:
+          paths:
+            - ~/.m2
+          key: v1-dependencies-{{ checksum "pom.xml" }}
+
+      # Executing build
+      - run:
+          name: Creating artifacts and executing tests
+          command: mvn -B clean package
+
+      - run:
+          name: Collecting artifacts and test reports
+          when: always
+          command: |
+            mkdir -p results/artifacts results/reports
+            cp -a target/surefire-reports results/reports/junit
+            cp -a target/*.jar results/artifacts
+
+      - store_test_results:
+          path: results/reports
+
+      - store_artifacts:
+          path: results
```

Comment:
Probably it should be results/artifacts. In such case "destination" is not required.

Reply:
Or you treat test reports as artifacts?

Reply:
Yep, I would allow the reports to be available - because they contain stdout & other info that is not showed by circleci right now.